test/azure/terraform_azure_vm_example_test.go

```diff
@@ -0,0 +1,191 @@
+	resourceGroupName := terraform.Output(t, terraformOptions, "resource_group_name")
+	virtualMachineName := terraform.Output(t, terraformOptions, "vm_name")
+	expectedVNetName := terraform.Output(t, terraformOptions, "virtual_network_name")
+	expectedSubnetName := terraform.Output(t, terraformOptions, "subnet_name")
+	expectedPublicAddressName := terraform.Output(t, terraformOptions, "public_ip_name")
+	expectedNicName := terraform.Output(t, terraformOptions, "network_interface_name")
+	expectedAvsName := terraform.Output(t, terraformOptions, "availability_set_name")
+	expectedOSDiskName := terraform.Output(t, terraformOptions, "os_disk_name")
+	expectedDiskName := terraform.Output(t, terraformOptions, "managed_disk_name")
+
+	// Comment for PR, to be removed:
+	// Please let me know if there are too many tests or alternate examples, happy to reduce the
+	// complexity and amount of code to be maintained. I tried to illustrate different approaches
+	// we have used in various scenarios to illustrate the flexability of Terratest.
+
+	t.Run("Strategies", func(t *testing.T) {
+		// Check the VM Size directly.
+		actualVMSize := azure.GetVirtualMachineSize(t, virtualMachineName, resourceGroupName, subscriptionID)
+		assert.Equal(t, expectedVMSize, actualVMSize)
+
+		// Check the VM Size by reference alternate example.
+		// This strategy is beneficial when checking multiple properties by using one VM reference, avoiding
+		// multiple SDK calls.
+		vmRef := azure.GetVirtualMachine(t, virtualMachineName, resourceGroupName, subscriptionID)
+		actualVMSize = vmRef.HardwareProfile.VMSize
+		assert.Equal(t, expectedVMSize, actualVMSize)
+
+		// Check the VM Size by instance alternate example.
+		// This strategy is beneficial when checking multiple properties by using one VM instance and making
+		// calls against it with the added benefit of property checks and abstraction.
+		vmInstance := azure.GetVirtualMachineInstance(t, virtualMachineName, resourceGroupName, subscriptionID)
+		actualVMSize = vmInstance.GetVirtualMachineInstanceSize()
+		assert.Equal(t, expectedVMSize, actualVMSize)
+	})
+
+	t.Run("MultipleVMs", func(t *testing.T) {
+		// This is beneficial when multiple VMs in a Resource Group need to be tested at once.
+
+		// Check against all VM names in a Resource Group.
+		vmList := azure.ListVirtualMachinesForResourceGroup(t, resourceGroupName, subscriptionID)
+		assert.True(t, len(vmList) > 0)
+		assert.Contains(t, vmList, virtualMachineName)
+
+		// Get all VMs in a Resource Group by reference alternate example.
+		// This strategy is beneficial when checking multiple VMs & their properties by avoiding
+		// multiple SDK calls. The penalty for this approach is introducing direct references
+		// which need to be checked for nil for optional configurations.
+		vmsByRef := azure.GetVirtualMachinesForResourceGroup(t, resourceGroupName, subscriptionID)
+		assert.True(t, len(*vmsByRef) > 0)
```

Comment:
NIT: use `assert.Greater`, as it will show you both numbers when the assertion fails.

Reply:
Fixed.